client_end_user/shared/src/commonMain/kotlin/presentation/auth/signup/registration/RegistrationScreen.kt

```diff
@@ -0,0 +1,132 @@
+import androidx.compose.animation.slideOutVertically
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Icon
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.unit.dp
+import cafe.adriel.voyager.navigator.Navigator
+import com.beepbeep.designSystem.ui.composable.BPSnackBar
+import com.beepbeep.designSystem.ui.composable.BpButton
+import com.beepbeep.designSystem.ui.composable.BpTextField
+import com.beepbeep.designSystem.ui.theme.Theme
+import org.jetbrains.compose.resources.ExperimentalResourceApi
+import org.jetbrains.compose.resources.painterResource
+import presentation.auth.signup.registrationSubmit.RegistrationSubmitScreen
+import presentation.base.BaseScreen
+import presentation.composable.BpBrandBackgroundContainer
+import presentation.composable.HeadFirstCard
+import presentation.composable.modifier.noRippleEffect
+import resources.Resources
+
+class RegistrationScreen :
+    BaseScreen<RegistrationScreenModel, RegistrationUIState, RegistrationScreenEffect, RegistrationInteractionListener>() {
+
+    @Composable
+    override fun Content() {
+        initScreen(getScreenModel())
+    }
+
+    @OptIn(ExperimentalMaterial3Api::class, ExperimentalResourceApi::class)
+    @Composable
+    override fun onRender(
+        state: RegistrationUIState,
+        listener: RegistrationInteractionListener
+    ) {
+        BpBrandBackgroundContainer {
+            Row(
+                Modifier.height(56.dp).fillMaxWidth().align(Alignment.TopCenter)
+                    .padding(horizontal = 16.dp, vertical = 8.dp),
+                verticalAlignment = Alignment.CenterVertically
+            ) {
+                Box(
+                    Modifier.size(40.dp).clip(RoundedCornerShape(Theme.radius.medium))
+                        .background(color = Theme.colors.surface)
+                        .noRippleEffect { listener.onBackButtonClicked() },
+                    contentAlignment = Alignment.Center
+                ) {
+                    Icon(
+                        painter = painterResource(Resources.images.icBack),
+                        contentDescription = null,
+                        tint = Theme.colors.primary,
+                        modifier = Modifier.size(16.dp)
+                    )
+                }
+            }
+            HeadFirstCard(
```

Comment:
HeadFirstCard isn't a clean code name "DD

Reply:
in another branch cause there is branches depend on this composable function